src/libs/kata-types/src/annotations/mod.rs

```diff
@@ -12,3 +23,754 @@ pub mod crio;
+pub const KATA_ANNO_CONF_HYPERVISOR_VIRTIO_FS_CACHE_SIZE: &str =
+    "io.katacontainers.config.hypervisor.virtio_fs_cache_size";
+/// A sandbox annotation to pass options to virtiofsd daemon.
+pub const KATA_ANNO_CONF_HYPERVISOR_VIRTIO_FS_EXTRA_ARGS: &str =
+    "io.katacontainers.config.hypervisor.virtio_fs_extra_args";
+/// A sandbox annotation to specify as the msize for 9p shares.
+pub const KATA_ANNO_CONF_HYPERVISOR_MSIZE_9P: &str = "io.katacontainers.config.hypervisor.msize_9p";
+
+// Runtime related annotations
+/// Prefix for Runtime configurations.
+pub const KATA_ANNO_CONF_RUNTIME_PREFIX: &str = "io.katacontainers.config.runtime.";
+/// A sandbox annotation that determines if seccomp should be applied inside guest.
+pub const KATA_ANNO_CONF_DISABLE_GUEST_SECCOMP: &str =
+    "io.katacontainers.config.runtime.disable_guest_seccomp";
+/// A sandbox annotation that determines if pprof enabled.
+pub const KATA_ANNO_CONF_ENABLE_PPROF: &str = "io.katacontainers.config.runtime.enable_pprof";
+/// A sandbox annotation that determines if experimental features enabled.
+pub const KATA_ANNO_CONF_EXPERIMENTAL: &str = "io.katacontainers.config.runtime.experimental";
+/// A sandbox annotaion that determines how the VM should be connected to the the container network
+/// interface.
+pub const KATA_ANNO_CONF_INTER_NETWORK_MODEL: &str =
+    "io.katacontainers.config.runtime.internetworking_model";
+/// SandboxCgroupOnly is a sandbox annotation that determines if kata processes are managed only in sandbox cgroup.
+pub const KATA_ANNO_CONF_SANDBOX_CGROUP_ONLY: &str =
+    "io.katacontainers.config.runtime.sandbox_cgroup_only";
+/// A sandbox annotation that determines if create a netns for hypervisor process.
+pub const KATA_ANNO_CONF_DISABLE_NEW_NETNS: &str =
+    "io.katacontainers.config.runtime.disable_new_netns";
+/// A sandbox annotation to specify how attached VFIO devices should be treated.
+pub const KATA_ANNO_CONF_VFIO_MODE: &str = "io.katacontainers.config.runtime.vfio_mode";
+
+/// A helper structure to query configuration information by check annotations.
+#[derive(Debug, Default, Deserialize)]
+pub struct Annotation {
+    annotations: HashMap<String, String>,
+}
+
+impl From<HashMap<String, String>> for Annotation {
+    fn from(annotations: HashMap<String, String>) -> Self {
+        Annotation { annotations }
+    }
+}
+
+impl Annotation {
+    /// Create a new instance of [`Annotation`].
+    pub fn new(annotations: HashMap<String, String>) -> Annotation {
+        Annotation { annotations }
+    }
+
+    /// Deserialize an object from a json string.
+    pub fn deserialize<T>(path: &str) -> Result<T>
+    where
+        for<'a> T: Deserialize<'a>,
+    {
+        let f = BufReader::new(File::open(path)?);
+        Ok(serde_json::from_reader(f)?)
+    }
+
+    /// Get an immutable reference to the annotation hashmap.
+    pub fn get_annotation(&self) -> &HashMap<String, String> {
```

Comment:
Another nit: `get_annotations`, otherwise it sounds like a getter for a single one.

Reply:
fixed